app/Http/Controllers/Api/AssetsController.php

```diff
@@ -542,6 +543,15 @@ public function store(StoreAssetRequest $request): JsonResponse
         $asset->fill($request->validated());
         $asset->user_id    = Auth::id();
 
+		/**
+        * rule of set up EOL date and explicit status during create asset
+		*/
+		if ($request->has('asset_eol_date')) {
```

Comment:
I'm not sure about this, I think I'd like to still do a `diffInMonths()` to make sure it _is_ explicit, unless they are also sending `eol_explicit` in the request.

Reply:
For create new assets via API (`store()` method), I'm think it is ok, not required to use function `diffInMonths()` addtionally for check. In this case we leave decision for user. Using attribut `asset_eol_date` user want to set eol date manully, out of the model eol rule. This same if is used `eol_explicit` attribut with value `1`. In both cases it's intentionally user's action.
Otherwaise applyed is asset model eol rule if purchase date filled and model eol rate greaten then zero.